v-next/hardhat/src/internal/builtin-plugins/solidity-test/reporter.ts

```diff
@@ -14,47 +19,119 @@ import chalk from "chalk";
 export async function* testReporter(
   source: TestEventSource,
 ): TestReporterResult {
-  let testResultCount = 0;
-  let successCount = 0;
-  let failureCount = 0;
-  let skippedCount = 0;
+  let runTestCount = 0;
+  let runSuccessCount = 0;
+  let runFailureCount = 0;
+  let runSkippedCount = 0;
+  let runDuration = 0n;
+
+  const failures: TestResult[] = [];
 
   for await (const event of source) {
     switch (event.type) {
       case "suite:result": {
         const { data: suiteResult } = event;
-        for (const testResult of suiteResult.testResults) {
-          testResultCount++;
-          let name = suiteResult.id.name + " | " + testResult.name;
-          if ("runs" in testResult?.kind) {
-            name += ` (${testResult.kind.runs} runs)`;
+        const suiteTestCount = suiteResult.testResults.length;
+
+        if (suiteTestCount === 0) {
```

Comment:
When can this happen? An empty test contract, for example?

Reply:
Yes, exactly or an abstract one only.